pkg/operator/controller/controller_dns_service.go

```diff
@@ -21,16 +23,21 @@ func (r *reconciler) ensureDNSService(dns *operatorv1.DNS, clusterIP string, dae
 	if err != nil {
 		return nil, err
 	}
-	if current != nil {
-		return current, nil
-	}
-
 	desired := desiredDNSService(dns, clusterIP, daemonsetRef)
-	if err := r.client.Create(context.TODO(), desired); err != nil {
-		return nil, fmt.Errorf("failed to create dns service: %v", err)
+
+	switch {
+	case desired != nil && current == nil:
+		if err := r.client.Create(context.TODO(), desired); err != nil {
+			return nil, fmt.Errorf("failed to create dns service: %v", err)
+		}
+		logrus.Infof("created dns service: %s/%s", desired.Namespace, desired.Name)
+		return desired, nil
```

Comment:
Could delete this line and fall out to the `return r.currentDNSService(dns)` statement below, which would return the actual API object.  Our `ensure` methods are inconsistent with respect to which value they return though.

Reply:
Fixed